tests/dataset_test.py

```diff
@@ -0,0 +1,141 @@
+from qri import dataset
+from qri import loader
+import mock_loader
+import unittest
+
+# TODO - move these to a better place?
+LIST_OBJ = {
+    'username': 'me',
+    'profileID': 'Qm.profileID',
+    'name': 'first_dataset',
+    'path': '/ipfs/Qm.path',
+    'bodySize': 1,
+    'bodyRows': 2,
+    'numErrors': 3,
+    'bodyFormat': 'json',
+    'commitTime': '2020-08-24T20:00:32.171549113Z',
+    'fsiPath': '/home/my_user/datasets/first_dataset'
+}
+
+GET_OBJ = {
+    "bodyPath": "/ipfs/Qm.bodyPath",
+    "commit": {
+        "author": {"id": "Qm.commit.author.id"},
+        "message": "commit.message",
+        "timestamp": "2020-01-02T03:04:05.124963898Z",
+    },
+    "meta": {
+        "description": "meta.description",
+        "keywords": ["meta.keyword1", "meta.keyword2"],
+    },
+    "name": "first_dataset",
+    "path": "/ipfs/Qm.path",
+    "peername": "me",
+    "previousPath": "/ipfs/Qm.previousPath",
+    "readme": {
+        "scriptBytes": "VEVTVCBSRUFETUUK",
+    },
+    "structure": {
+        "checksum": "Qm.structure.checksum",
+        "entries": 3,
+        "format": "csv",
+        "schema": {
+            "items": {
+                "items": [
+                    {"title": "field1", "type": "string"},
+                    {"title": "field2", "type": "integer"},
+                ],
+                "type": "array",
+            },
+            "type": "array",
+        },
+    },
+}
+
+
+class DatasetTests(unittest.TestCase):
+    def test_init_from_list(self):
+        loader.set_instance(mock_loader.DatasetMockLoader())
+        ds = dataset.Dataset(LIST_OBJ)
+        self.assertEqual(ds.username, 'me')
+        self.assertEqual(ds.name, 'first_dataset')
+        self.assertEqual(ds.profile_id, 'Qm.profileID')
+        self.assertEqual(ds.path, '/ipfs/Qm.path')
+        self.assertFalse(ds._is_populated)
+
+    def test_init_from_get(self):
+        loader.set_instance(mock_loader.DatasetMockLoader())
+        ds = dataset.Dataset(GET_OBJ)
+        self.assertEqual(ds.username, 'me')
+        self.assertEqual(ds.name, 'first_dataset')
+        self.assertEqual(ds.path, '/ipfs/Qm.path')
+
+        self.assertEqual(ds.body_path_value, "/ipfs/Qm.bodyPath")
+        self.assertEqual(ds.previous_path_value, "/ipfs/Qm.previousPath")
+
+        # Checking a subset of the possible fields for brevity
+
+        # TODO - should I test every field? I realized it would get pretty verbose but I can.
```

Comment:
I think this is quite thorough enough, as long as it's clearly showing that the codepath in question is working as intended. You can remove the TODO and related questions.